### PR TITLE
[sysrst_ctrl] Update the debouncer behavior

### PR DESCRIPTION
**Pull request by msfschaffner**
This makes the debounce behavior more similar to the initial design, as it corresponds better to the intended programming model. That means that the debouncer logic is implemented using an FSM + counter, rather than using an instance of `prim_filter`.

Design-wise however, this still uses the approach that reuses one common detector module for all debounce and detect use cases within `sysrst_ctrl`.

The PR also adds an FPV target to prove embedded assertions.

-------------------------------
@cindychip: I have added you mainly for the FPV part.

-------------------------------

CC @patel-madhuri: apologies for the back and forth on this. I hope this is the last alignment that is needed.

### CODE REVIEW

File: hw/ip/sysrst_ctrl/rtl/sysrst_ctrl_detect.sv
Comment (tjaychen) [inline hw/ip/sysrst_ctrl/rtl/sysrst_ctrl_detect.sv:69]: i might have missed something, but is `trigger_i` separately synchronized already outside the module?

**Review by tjaychen — APPROVED** [review]
lgtm with one question.

File: hw/ip/sysrst_ctrl/rtl/sysrst_ctrl_detect.sv
Comment (msfschaffner) [reply hw/ip/sysrst_ctrl/rtl/sysrst_ctrl_detect.sv:69]: Yes, there is a bank of shared CDCs at the top-level of this IP:
https://github.com/lowRISC/opentitan/blob/a463e17f1b7aadea1b032fd3f01e4492821d957b/hw/ip/sysrst_ctrl/rtl/sysrst_ctrl.sv#L116-L131

**Comment by msfschaffner** [comment]
Thanks @tjaychen. I'll hold off merging this until we clarified https://github.com/lowRISC/opentitan/issues/10663 - just to be sure this is implemented as intended.

**Review by cindychip — COMMENTED** [review]
Thanks! Assertions + configs LGTM!

**Comment by msfschaffner** [comment]
> Thanks @tjaychen. I'll hold off merging this until we clarified #10663 - just to be sure this is implemented as intended.

EDIT: I merge this in now in order to unblock DV. However we should still double check on #10663 whether the current implementation does indeed fulfill our requirements.

**Pull request merged**
